components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -212,424 +171,357 @@ public Map<String, Integer> getActionsCountPerType(Integer tenantId) throws Acti
-                                    property.getValue());
-                            statement.setString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_PROPERTY_NAME,
-                                    property.getKey());
+                        for (Map.Entry<String, String> property : actionProperties.entrySet()) {
                             statement.setString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_UUID, actionId);
                             statement.setInt(ActionMgtSQLConstants.Column.TENANT_ID, tenantId);
+                            statement.setString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_PROPERTY_NAME,
+                                    property.getKey());
+                            statement.setString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_PROPERTY_VALUE,
+                                    property.getValue());
                             statement.addBatch();
                         }
                     }, null);
-                return null;
-            });
-        } catch (TransactionException e) {
-            throw ActionManagementUtil.handleServerException(
-                    ActionMgtConstants.ErrorMessages.ERROR_WHILE_UPDATING_ENDPOINT_PROPERTIES, e);
-        }
+            return null;
+        });
     }
 
     /**
-     * Update Action Status.
+     * Get Action properties by ID.
      *
-     * @param actionType Action Type.
-     * @param actionId   UUID of the Action.
-     * @param status     Action status to be updated.
-     * @param tenantId   Tenant ID.
-     * @throws ActionMgtException If an error occurs while updating the Action status.
+     * @param actionId UUID of the created Action.
+     * @param tenantId Tenant ID.
+     * @return A map of action properties, including any additional data based on action type.
+     * @throws ActionMgtException If an error occurs while retrieving action properties from the database.
      */
-    private Action changeActionStatus(String actionType, String actionId, String status, Integer tenantId)
-            throws ActionMgtException {
+    private Map<String, String> getActionPropertiesFromDB(String actionId, Integer tenantId) throws ActionMgtException {
 
         NamedJdbcTemplate jdbcTemplate = new NamedJdbcTemplate(IdentityDatabaseUtil.getDataSource());
+        Map<String, String> actionEndpointProperties = new HashMap<>();
         try {
-            jdbcTemplate.executeUpdate(ActionMgtSQLConstants.Query.CHANGE_ACTION_STATUS,
+            jdbcTemplate.executeQuery(ActionMgtSQLConstants.Query.GET_ACTION_ENDPOINT_INFO_BY_ID,
+                (resultSet, rowNumber) -> {
+                    actionEndpointProperties.put(
+                            resultSet.getString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_PROPERTY_NAME),
+                            resultSet.getString(ActionMgtSQLConstants.Column.ACTION_ENDPOINT_PROPERTY_VALUE));
+                    return null;
+                },
```

Comment:
So, these are now no longer just endpoint related properties right ? These are action related properties. Therefore let's rename the constants for readability

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6119/commits/dab9dd75643167cc04aa5d35aec2f044d59a58d4